base/file.jl

```diff
@@ -420,19 +420,20 @@ const temp_prefix = "jl_"
 
 if Sys.iswindows()
 
-function _isroot(path::String)
+function ispathroot(path::String)
     m = match(r"^([^\\]+:|\\\\[^\\]+\\[^\\]+|\\\\\?\\UNC\\[^\\]+\\[^\\]+|\\\\\?\\[^\\]+:|)(.*)$", path)
-    m.captures[2] == "\\" ? true : false
+    (m.captures[2] == "\\" || isempty(m.captures[2])) ? true : false
 end
 
 function tempdir()
     temppath = Vector{UInt16}(undef, 32767)
     lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
     windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath,lentemppath)
+    resize!(temppath,lentemppath-1) # ignore trailing `\`
```

Comment:
Shouldn't it check that there is a trailing `\` first?  e.g.
```jl
resize!(temppath,lentemppath - (temppath[lentemppath]==0x005c))
```

Reply:
According to their docs https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-gettemppathw
the returned string ends with a backslash.

Reply:
I like how their docs say that it ends with a backslash and immediately contradicts itself with the example: ``` The returned string ends with a backslash, for example, "C:\TEMP". ```

Reply:
I know, someone at MSFT needs to fix that....    (I tested several combinations and the function always adds backslash...)

Reply:
Apparently already reported https://github.com/MicrosoftDocs/feedback/issues/898